import { NextFunction, Request, RequestHandler, Response } from "express";
import globalErrorHandaler from "./globalErrorhandler";

const catchAsync = (fn: RequestHandler) => {
  return (req: Request, res: Response, next: NextFunction) => {
    Promise.resolve(fn(req, res, next)).catch((error) => next(error));
  };
};

export default catchAsync;
